Calculator/Program.cs

```diff
@@ -0,0 +1,238 @@
+                    }
+                }
+                else
+                {
+                    cleanNum1 = firstNumber;
+                }
+
+                // Ask the user to type the second number.
+                if (double.IsNaN(secondNumber))
+                {
+                    Console.Write("Type second number, and then press Enter: ");
+                    numInput2 = Console.ReadLine();
+
+                    cleanNum2 = 0;
+                    while (!double.TryParse(numInput2, out cleanNum2))
+                    {
+                        Console.Write("This is not valid input. Please enter an integer value: ");
+                        numInput2 = Console.ReadLine();
+                    }
+                } 
+                else
+                {
+                    cleanNum2 = secondNumber;
+                }
+
+                // Ask the user to choose an operator.
+                Console.WriteLine("Choose an operator from the following list:");
+                Console.WriteLine("Please note, only the first number will be used for square root,");
+                Console.WriteLine("10x, sine, cosine, tangent and cotangent.");
+                Console.WriteLine("\ta - Add");
+                Console.WriteLine("\ts - Subtract");
+                Console.WriteLine("\tm - Multiply");
+                Console.WriteLine("\td - Divide");
+                Console.WriteLine("\tp - Power");
+                Console.WriteLine("\tr - Square Root");
+                Console.WriteLine("\tx - 10x");
+                Console.WriteLine("\tsin - Sine");
+                Console.WriteLine("\tcos - Cosine");
+                Console.WriteLine("\ttan - Tangent");
+                Console.WriteLine("\tcot - Cotangent");
+                Console.Write("Your option? ");
+
+                string op = Console.ReadLine();
+
+                try
+                {
+                    result = calculator.DoOperation(cleanNum1, cleanNum2, op);
+                    if (double.IsNaN(result))
+                    {
+                        Console.WriteLine("This operation will result in a mathematical error.\n");
+                    }
+                    else Console.WriteLine("Your result: {0:0.##}\n", result);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("Oh no! An exception occurred trying to do the math.\n - Details: " + e.Message);
+                }
+                firstNumber = double.NaN;
+                secondNumber = double.NaN;
+
+                Console.WriteLine("------------------------\n");
+
+                // Wait for the user to respond before closing.
+                Console.Write("Press 'n' to close the app, 'm' for main menu or press any other key and Enter to continue: ");
+                
+                var input = Console.ReadLine().ToLower();
+                
+                if (input.Equals("n")) Environment.Exit(0);
```

Comment:
After closing the application, I realized that the json file is not complete (see the second image). This is because you used the `Environment.Exit(0)` method to terminate the application. My suggestion here is to get rid of that `Environment.Exit(0)` and just send the user back to main menu. they should choose if they want to quit or check history. Also at the main menu I have a suggestion you should check it out

![image](https://github.com/TheCSharpAcademy/CodeReviews.Console.Calculator/assets/57032138/c79b79b3-21b9-4a0d-910c-fa7d9a87c4c1)

![image](https://github.com/TheCSharpAcademy/CodeReviews.Console.Calculator/assets/57032138/d0c6e18e-3fbb-45c9-a205-35ce75047796)